spec/controllers/api/v1/profiles_controller_spec.rb

```diff
@@ -14,34 +15,119 @@
+      expect(data[1]).to include('id' => profile2.id)
     end
 
     it 'should filter published profiles by id' do
       get :index, ids: [profile1.id], format: :json
 
-      expect(response.body).to include('{"id":' + profile1.id.to_s + ',"firstname"')
-      expect(response.body).not_to include('{"id":' + profile2.id.to_s + ',"firstname"')
+      expect(data[0]).to include('id' => profile1.id)
+      expect(data[1]).to be_nil
     end
 
     it 'should not show unpublished profiles' do
       get :index, ids: [unpublished_profile.id], format: :json
-
-      expect(response.body).to eq '[]'
+      expect(data[0]).to be_nil
     end
   end
 
   describe 'show action' do
     it 'should respond to json request for one profile' do
       http_login('horst', '123')
       get :show, id: profile1.id, format: 'json'
-      expect(response.body).to include('{"id":' + profile1.id.to_s + ',"firstname":"Factory","lastname":"Girl"')
+      expect(data).to include('id' => profile1.id, 'firstname' => 'Factory', 'lastname' => 'Girl')
     end
 
     it 'should deny json requests without a login' do
       get :show, id: profile1.id, format: 'json'
       expect(response.status).to eq(401)
     end
   end
+
+  describe 'containing attributes' do
+    before do
+      http_login('horst', '123')
+    end
+
+    it 'should contain attribute id' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['id']).to eq(profile1.id)
+    end
+
+    it 'should contain attribute firstname' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['firstname']).to eq('Factory')
+    end
+
+    it 'should contain attribute lastname' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['lastname']).to eq('Girl')
+    end
+
+    it 'should contain attribute city' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['city']).to eq('New York')
+    end
+
+    it 'should contain attribute country' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['country']).to eq('Unicornland')
+    end
+
+    it 'should contain attribute twitter' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['twitter']).to eq('factorygirl')
+    end
+
+    it 'should contain attribute picture' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['picture']).to eq(nil)
+    end
+
+    it 'should contain attribute created_at' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['created_at']).to eq(profile1.created_at.as_json)
+    end
+
+    it 'should contain attribute updated_at' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['updated_at']).to eq(profile1.updated_at.as_json)
+    end
+
+    it 'should contain attribute medialinks' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['medialinks']).to eq([])
```

Comment:
One day we have to add medialinks etc there to check them as well. but for now I guess its ok.